### PR TITLE
Refactored Pearson code to separate functionality

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?

None

#### What's this PR do?

This PR refactors the Pearson-specific code into it's own module within the exams application. Since we'll be adding more Pearson code in the coming weeks a single file for all Pearson code isn't a good thing.

#### Where should the reviewer start?

exams/pearson/*.py

#### How should this be manually tested?

This is a refactor-only PR, so passing tests should be sufficient since we have good test coverage in this application.

### CODE REVIEW

File: exams/pearson/sftp_test.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+class PearsonUploadTest(SimpleTestCase):
+    """
+    Tests for Pearson upload
+    """
+
+    def test_upload_tsv(self):
+        """
+        Tests that upload uses the correct settings values
+        """
+        with self.settings(**EXAMS_SFTP_SETTINGS), patch('pysftp.Connection') as connection_mock:
```
Comment (gsidebo) [inline exams/pearson/sftp_test.py]: the settings override and connection patch is repeated in each test case in this class. i tried this with a few changes to get rid of the redundancy and the interruption of the `with` statements. this worked for me:

```python
@ddt
@override_settings(**EXAMS_SFTP_SETTINGS)
@patch('pysftp.Connection')
class PearsonUploadTest(SimpleTestCase):
    """
    Tests for Pearson upload
    """
    def test_upload_tsv(self, connection_mock):
        """
        Tests that upload uses the correct settings values
        """
        sftp.upload_tsv(EXAMS_SFTP_FILENAME)
        connection_mock.assert_called_once_with(
            host=EXAMS_SFTP_HOST,
            port=int(EXAMS_SFTP_PORT),
            username=EXAMS_SFTP_USERNAME,
            password=EXAMS_SFTP_PASSWORD,
            cnopts=ANY,
        )

        ftp_mock = connection_mock.return_value.__enter__.return_value
        ftp_mock.cd.assert_called_once_with(EXAMS_SFTP_UPLOAD_DIR)
        ftp_mock.put.assert_called_once_with(EXAMS_SFTP_FILENAME)

    @data(
        SSHException(),
        ConnectionException('localhost', 22),
    )
    def test_retryable_exceptions(self, expected_exc, connection_mock):
        """
        Test that if {exc_cls} is raised that it results in a RetryableSFTPException
        """
        connection_mock.side_effect = expected_exc
        with self.assertRaises(RetryableSFTPException) as cm:
            sftp.upload_tsv(EXAMS_SFTP_FILENAME)

        assert isinstance(cm.exception, RetryableSFTPException)
        assert cm.exception.__cause__ == expected_exc

    @data(
        'EXAMS_SFTP_HOST',
        'EXAMS_SFTP_PORT',
        'EXAMS_SFTP_USERNAME',
        'EXAMS_SFTP_PASSWORD',
        'EXAMS_SFTP_UPLOAD_DIR',
    )
    def test_upload_tsv_fails_if_settings_missing(self, settings_key, connection_mock):
        """
        Tests that upload raises ImproperlyConfigured if settings.{0} is not set
        """
        kwargs = {settings_key: None}

        with self.settings(**kwargs):
            with self.assertRaises(ImproperlyConfigured):
                sftp.upload_tsv('file.tsv')

            connection_mock.assert_not_called()
```

File: exams/pearson/sftp.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    Upload the given TSV files to the remote
+
+    Args:
+        file_path (str): absolute path to the file to be uploaded
+    """
+    for key in PEARSON_UPLOAD_REQUIRED_SETTINGS:
+        if getattr(settings, key) is None:
+            raise ImproperlyConfigured(
+                "The {} setting is required".format(key)
+            )
```
Comment (rhysyngsun) [inline exams/pearson/sftp.py]: Implemented in this PR.

File: exams/pearson/sftp_test.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        assert isinstance(cm.exception, RetryableSFTPException)
+        assert cm.exception.__cause__ == expected_exc
+
+    @data(
+        'EXAMS_SFTP_HOST',
+        'EXAMS_SFTP_PORT',
+        'EXAMS_SFTP_USERNAME',
+        'EXAMS_SFTP_PASSWORD',
+        'EXAMS_SFTP_UPLOAD_DIR',
+    )
```
Comment (rhysyngsun) [inline exams/pearson/sftp_test.py]: I ended up unpacking `PEARSON_UPLOAD_REQUIRED_SETTINGS` into `@data` instead, which gets the same job done but preserves `ddt`'s parameterization of this test.

File: exams/pearson/upload_test.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        """
+        kwargs = {settings_key: None}
+
+        with self.settings(**kwargs), patch('pysftp.Connection') as connection_mock:
+            with self.assertRaises(ImproperlyConfigured) as cm:
+                upload.upload_tsv('file.tsv')
+
+            connection_mock.assert_not_called()
+
+            print(cm.exception)
```
Comment (gsidebo) [inline exams/pearson/upload_test.py]: do we want this `print` statement (and the linebreak above it)?

Comment (rhysyngsun) [inline exams/pearson/upload_test.py]: Oops, good catch.

**Pull request merged**